docs/book/how-to/build-pipelines/dynamically-generate-steps-artifacts.md

```diff
@@ -0,0 +1,105 @@
+---
+description: How to dynamically generate steps and artifacts in your pipelines.
+---
+
+# Dynamically Generating Steps and Artifacts
```

Comment:
I think maybe the framing of this is "Dynamically assign artifact names" . The dynamic step generation just happens to be a byproduct